### ISSUE TITLE
After AddWorkbookPart() I recieved "Cannot retrieve parts of writeonly container" Exception from System.IO.Packaging

### ISSUE THREAD

**Issue by Reules** [dotnet/Open-XML-SDK#1809]
**Describe the bug**
Recently I updated OpenXML from 2.18.0 to 3.1.0.
The function AddWorkbookPart() didn't work anymore. 
Always get the System.IO.IOException "Cannot retrieve parts of writeonly container".

**Screenshots**
![Image](https://github.com/user-attachments/assets/59a63504-4b01-4a79-b859-ed0a99cfdf9e)

Anyone knows the problem?

**Comment by mikeebowen** [comment]
Hi @Reules, can you share the code that causes this error? 

Thanks!

**Comment by mikeebowen** [comment]
Hi @Reules,

I am using v3.1.1 and the following code with `AddWorkbookPart` creates an xlsx with no errors.

```csharp
using DocumentFormat.OpenXml;
using DocumentFormat.OpenXml.Packaging;
using DocumentFormat.OpenXml.Spreadsheet;

string xlsx = @"C:\some\dir\AddWorkbookPart.xlsx";

// Create a workbook document by supplying the file path.
// By default, AutoSave = true, Editable = true, and Type = xlsx.
using (var spreadsheetDocument = SpreadsheetDocument.Create(xlsx, SpreadsheetDocumentType.Workbook))
{
    // Add a WorkbookPart to the document.
    WorkbookPart workbookPart = spreadsheetDocument.AddWorkbookPart();
    workbookPart.Workbook = new Workbook();

    // Add a WorksheetPart to the WorkbookPart.
    WorksheetPart worksheetPart = workbookPart.AddNewPart<WorksheetPart>();
    worksheetPart.Worksheet = new Worksheet(new SheetData());

    // Add Sheets to the workbook.
    Sheets? sheets = spreadsheetDocument?.WorkbookPart?.Workbook.AppendChild<Sheets>(new Sheets());

    // Append a new worksheet and associate it with the workbook.
    Sheet? sheet = new Sheet()
    {
        Id = spreadsheetDocument?.WorkbookPart?.GetIdOfPart(worksheetPart),
        SheetId = 1,
        Name = "MyNewSheet"
    };

    if (sheets is not null && sheet is not null)
    {
        sheets.Append(sheet);
    }
}
```

**Comment by mikeebowen** [comment]
Hi @Reules, Are you somehow opening the file as write only?

**Comment by Reules** [comment]
Hi @mikeebowen Thanks for the reply, yes, before the AddWorkbookPart, i used 
var package = Package.Open(fs, FileMode.Create, FileAccess.Write), which did't cause any problem with Openxml 2.18.0
Maybe this was fixed in Openxml 3.1.0
Thank you for your help

**Comment by mikeebowen** [comment]
In System.IO.Packaging.Package, the `GetRelationshipsHelper` method calls a method `ThrowIfWriteOnly`, which throws `IOException(System.SR.CannotRetrievePartsOfWriteOnlyContainer)` if it is write only. The git commit for this is from 9 years ago, so I don't know how it works in 2.18.0, but it shouldn't. View it on GitHub here: [GetRelationshipsHelper method](https://github.com/dotnet/runtime/blob/07e9313bb3f2d1616f2ea722e92bc5b7573a061d/src/libraries/System.IO.Packaging/src/System/IO/Packaging/Package.cs#L1198).

Since what you're seeing is the expected behavior, I will close this as by design.
